src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt

```diff
@@ -45,12 +46,14 @@ class ChainAdapter(
 ) : Closeable {
 
     private val connectionFactory = ConnectionFactory()
-    private val lastReadBlock = AtomicLong()
+
     private val publishUnreadLatch = CountDownLatch(1)
     private val subscriberExecutorService = createPrettySingleThreadPool(
         CHAIN_ADAPTER_SERVICE_NAME, "iroha-chain-subscriber"
     )
 
+    private var lastReadBlock = AtomicReference<BigInteger>()
```

Comment:
I'd rather use `val`. I also suggest you set BigInteger.ZERO as a default value.